extern/acts/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+# TRACCC library, part of the ACTS project (R&D line)
+#
+# (c) 2021 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+# CMake include(s).
+cmake_minimum_required( VERSION 3.11 )
+include( FetchContent )
+
+# Tell the user what's happening.
+message( STATUS "Building Acts as part of the TRACCC project" )
+
+# Declare where to get Acts from.
+set( TRACCC_ACTS_SOURCE
+   "URL;https://github.com/krasznaa/acts/archive/refs/tags/v10.0.1.tar.gz;URL_MD5;fcd2f2f7b5810e34b8faa12634043be3"
+   CACHE STRING "Source for Acts, when built as part of this project" )
+mark_as_advanced( TRACCC_ACTS_SOURCE )
+FetchContent_Declare( Acts ${TRACCC_ACTS_SOURCE} )
+
+# Options used in the build of Acts.
+set( ACTS_SETUP_EIGEN3 FALSE CACHE BOOL
+   "Do not set up Eigen in the Acts code, we do it in this project" )
```

Comment:
Are we certain that Eigen3 is always set up in traccc? I thought it was possible to disable that, no?

Reply:
It is. But by default it **is** built by traccc. So by default Acts should get it from the same place.

If you want to use another version of Eigen than what traccc is building, you can change all of these defaults. It will take modifying multiple flags, which I admit can get a little tiring. But all the flags are now there to allow any setup that you can think of.

Remember, all of these are cache variables. They can all be overridden from the command line if necessary.

Reply:
These being cache variables is all well and good, but doesn't this mean that you can quite easily break the build just by selecting a non-default way of building Eigen? I would think that the way Acts builds Eigen should be informed by how traccc does it - rather than just disabling the build in Acts all together. Sure, this is unlikely to cause problems in real-world cases and I'm fine to accept this as is, but it feels to me like we're putting together a Jenga tower here, where one (potentially sane) configuration change can break the entire build. :fearful:

Reply:
Right now, without https://github.com/acts-project/acts/pull/1071, Acts sets up Eigen with this code:

```cmake
if (ACTS_USE_SYSTEM_EIGEN3)
  find_package(Eigen3 ${_acts_eigen3_version} REQUIRED CONFIG)
else()
  add_subdirectory(thirdparty/eigen3)
endif()
```

https://github.com/acts-project/acts/blob/main/CMakeLists.txt#L154-L158

If I didn't make any changes in Acts, this would force us to always build Eigen before building traccc. The build of Eigen could not be done at the same time as this project. (Since Acts does not use a new enough version of Eigen for our use case.)

Yes. With this many cache variables defined, you can very much come up with non-functional setups. Maybe we can improve on that in the future. But I'm getting the sense that you don't understand what my thinking is with all these cache variables. I guess we should talk on Monday over Skype about this...